packages/git/src/browser/git-repository-provider.ts

```diff
@@ -56,6 +78,8 @@ export class GitRepositoryProvider {
         await this.refresh();
     }
 
+    protected lazyRefresh = debounce(() => this.refresh(), 1000);
```

Comment:
We'are trying to explicitly state type information for the API fields, properties, methods, and functions. Please add them too.

I would say, exposing the `Cancelable` type is not necessary in this case, but instead of writing

```ts
protected lazyRefresh = debounce(() => this.refresh(), 1000);
```

please do

```ts
protected lazyRefresh: () => Promise<void> = debounce(() => this.refresh(), 1000);
```

Thanks!

Reply:
ok, I've added the type,though surely this is not API (it is protected) unless one considers almost everything to be API because extensions could extend WorkspaceService.